src/openms/source/QC/MQEvidenceExporter.cpp

```diff
@@ -0,0 +1,480 @@
+    for(Size i = 0; i < cmap.size(); ++i)
+    {
+        for(const auto & fh : cmap[i].getFeatures())
+        {
+            f_to_ci[fh.getUniqueId()] = i;
+        }
+    }
+    return f_to_ci;
+}
+
+/*
+ * Description:
+ * Checks if a PeptideIdentification of a Feature can be used
+ * for the Export
+ * Returns true for yes and false for no
+ * changes the pointer of the best hit to the best hit
+ * of the PeptideIdentifications of the Feature
+ */
+bool MQEvidence::exportFeaturePepID(
+        const ConsensusMap &cmap,
+        const Int64 c_feature_number,
+        const std::multimap<OpenMS::String, std::pair<OpenMS::Size,OpenMS::Size>> &UIDs,
+        const ProteinIdentification::Mapping &mp_f,
+        const Feature & f,
+        const PeptideHit* & best)
+{
+    const std::vector<PeptideIdentification> &pep_ids_f = f.getPeptideIdentifications();
+    if(pep_ids_f.empty())
+    {
+        return false;
+    }
+    std::pair<Size,const PeptideHit*> hit = getBestPeptideHit(pep_ids_f);
+    best = hit.second;
+    const PeptideIdentification & best_pep_id = pep_ids_f[hit.first];
+    String best_uid = PeptideIdentification::buildUIDFromPepID(best_pep_id, mp_f.identifier_to_msrunpath);
+    const auto range = UIDs.equal_range(best_uid);
+    for (auto it_pep = range.first; it_pep != range.second; ++it_pep)
+    {
+        if(c_feature_number != it_pep->second.first)
+        {
+            return false;
+        }
+        const String & seq = best->getSequence().toString();
+        const String & seq_comp = cmap[it_pep->second.first].getPeptideIdentifications()[it_pep->second.second].getHits()[0].getSequence().toString();
+        if(seq != seq_comp)
+        {
+            return false;
+        }
+    }
+    return true;
+}
+
+/*
+ * Description:
+ * export one Feature as one row in MQEvidence.txt
+ */
+bool MQEvidence::exportConsensusFeaturePepID(
+        const ConsensusMap &cmap,
+        const Int64 c_feature_number,
+        const PeptideHit* & best)
+{
+    if(c_feature_number == -1) //ConsensusFeature does not exist?
+    {
+        return false;
+    }
+    const std::vector<PeptideIdentification> &pep_ids_c = cmap[c_feature_number].getPeptideIdentifications();
+    if(pep_ids_c.empty()){
+        return false;
+    }
+    best = getBestPeptideHit(pep_ids_c).second;
+    return true;
+
+}
+
+/*
+ * Description:
+ * export Feature as one row in MQEvidence.txt
+ */
+void MQEvidence::exportRowFromFeature(
+        const Feature &f,
+        const ConsensusMap &cmap,
+        const Int64 c_feature_number,
+        const String &raw_file,
+        const std::multimap<String, std::pair<Size, Size>> &UIDs,
+        const ProteinIdentification::Mapping &mp_f)
+{
+    const PeptideHit * pep_hits_max; // the best hit referring to score
+    UInt64 pep_ids_size = 0;
+    String type;
+    if(exportFeaturePepID(cmap, c_feature_number, UIDs, mp_f, f, pep_hits_max))
+    {
+        pep_ids_size = f.getPeptideIdentifications().size();
```

Comment:
count identical PEPIDs (first PeptideHit)

Reply:
int count{};
for (Size i = 1; i < f.getPeptideIds().size(); ++i)
{
  if (f.getPeptideIDs()[i].getHits()[0].getSequence() == f.getPeptideIds()[0].getHits()[0]) ++count;
  else break;
}